src/js/trove/world.js

```diff
@@ -504,75 +460,70 @@ define(["js/runtime-util", "trove/image-lib", "trove/world-lib", "js/ffi-helpers
         var makeFunction = runtime.makeFunction;
 
         return makeObject({
-          "provide": makeObject({
-            "big-bang": makeFunction(function(init, handlers) {
-              ffi.checkArity(2, arguments, "big-bang");
-              runtime.checkList(handlers);
-              var arr = ffi.toArray(handlers);
-              var initialWorldValue = init;
-              arr.map(function(h) { checkHandler(h); });
-              bigBang(initialWorldValue, arr);
-              ffi.throwMessageException("Internal error in bigBang: stack not properly paused and stored.");
-            }),
-            "on-tick": makeFunction(function(handler) {
-              ffi.checkArity(1, arguments, "on-tick");
-              runtime.checkFunction(handler);
-              return runtime.makeOpaque(new OnTick(handler, Math.floor(DEFAULT_TICK_DELAY * 1000)));
+          "provide-plus-types": makeObject({
+            "values": makeObject({
+              "big-bang": makeFunction(function(init, handlers) {
+                ffi.checkArity(2, arguments, "big-bang");
+                runtime.checkList(handlers);
+                var arr = ffi.toArray(handlers);
+                var initialWorldValue = init;
+                arr.map(function(h) { checkHandler(h); });
+                bigBang(initialWorldValue, arr);
+                ffi.throwMessageException("Internal error in bigBang: stack not properly paused and stored.");
```

Comment:
This is a fantastic idea.  I wonder if there's an easy way to wrap it in a helper/macro and make it a default thing that should be done for tail-position-pausing calls.